contracts/gateway/EIP20CoGateway.sol

```diff
@@ -867,11 +867,17 @@ contract EIP20CoGateway is GatewayBase {
             msg.value == bounty,
             "Payable amount should be equal to the bounty amount."
         );
+
         require(
             _amount > uint256(0),
             "Redeem amount must not be zero."
         );
 
+        require(
+            _amount > _gasPrice.mul(_gasLimit),
```

Comment:
For readability we could replace `_gasPrice.mul(_gasLimit)` with a variable that has a meaningful name, e.g. `maxPossibleReward`. However, in solidity land I am not sure how to balance readability and cost here...

Reply:
I would prefer to keep like this. 
I will add a comment in the code. 
We always struggled a lot with stack too deep issue in this function.

Reply:
Added a comment in both stake and redeem functions.

Reply:
👍